### PR TITLE
Profile update custom operation

### PR DESCRIPTION
**Pull request by gregory-latinier**

### CODE REVIEW

File: helpers/operations/custom-operations.js [hunk 1]
```diff
@@ -29,4 +29,9 @@ module.exports = [
     type: 'delegate_vesting_shares',
     params: ['delegator', 'delegatee'],
   },
+  {
+    operation: 'update_account',
```
Comment (bonustrack) [inline helpers/operations/custom-operations.js]: `update_account` is confusing with `account_update`. This helper is just for update profile you can name it `profile_update` instead.

File: helpers/operations/custom-operations.js [hunk 2]
```diff
@@ -29,4 +29,9 @@ module.exports = [
...
+    type: 'account_update',
+    params: ['account', 'memo_key'],
```
Comment (bonustrack) [inline helpers/operations/custom-operations.js]: `memo_key` should not be a param. We don't want to allow a change in user account `memo_key` using this helper. The memo_key should stay identical, we need to load user account for it.

Comment (gregory-latinier) [reply helpers/operations/custom-operations.js]: OK I'll correct it

File: helpers/operations/profile-update.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const changeCase = require('change-case');
+const customOperations = require('./custom-operations');
+const steem = require('steem');
+
+const optionalFields = ['account'];
+
+const parse = async (query) => {
+  const jsonMetadata = { profile: {} };
```
Comment (bonustrack) [inline helpers/operations/profile-update.js]: We should start from user jsonMetadata. User may have another field saved on his metadata. Think of someone with this for example:
```
{
  profile: {
    twitter: 'busy',
    name: 'Fabien'
  },
  someinfo: {
    hello: 1,
    world: 2
  }
}
```
When using sign update profile, with `name=Fab` parameter, i should not lose `someinfo` or `twitter`.

File: helpers/operations/profile-update.js [hunk 4]
```diff
@@ -5,7 +5,16 @@ const steem = require('steem');
 const optionalFields = ['account'];
 
 const parse = async (query) => {
-  const jsonMetadata = { profile: {} };
+  const accounts = await steem.api.getAccountsAsync([query.account]);
+  const account = accounts.find(a => a.name === query.account);
```
Comment (bonustrack) [inline helpers/operations/profile-update.js]: You can use this instead: `const [account] = await steem.api.getAccountsAsync([query.account]);`

Comment (gregory-latinier) [reply helpers/operations/profile-update.js]: The getAccountsAsync returns account with a like parameter if think. I encountered a case where the account username didn't match the actual search that's why I use the find here and in the validator too with the `userExists`

Comment (bonustrack) [reply helpers/operations/profile-update.js]: Can we have a validator for username then so this wont work: https://sc2-dev-pr-73.herokuapp.com/sign/profile-update?account=blablabla&about=test

Comment (bonustrack) [reply helpers/operations/profile-update.js]: There should be an error before review / sign similar to the others operation with `userExist` validator.

**Pull request merged**